core/chains/evm/headtracker/head_tracker.go

```diff
@@ -197,33 +242,34 @@ func (ht *headTracker) handleNewHead(ctx context.Context, head *evmtypes.Head) e
 		return errors.Wrapf(err, "failed to save head: %#v", head)
 	}
 
-	if prevHead == nil || head.Number > prevHead.Number {
-		promCurrentHead.WithLabelValues(ht.chainID.String()).Set(float64(head.Number))
+	if !prevHead.IsValid() || head.BlockNumber().Cmp(prevHead.BlockNumber()) == 1 {
+		promCurrentHead.WithLabelValues(ht.chainID.String()).Set(float64(head.BlockNumber().Int64()))
 
-		headWithChain := ht.headSaver.Chain(head.Hash)
-		if headWithChain == nil {
+		headWithChain := ht.headSaver.Chain(head.BlockHash())
+		if !headWithChain.IsValid() {
 			return errors.Errorf("HeadTracker#handleNewHighestHead headWithChain was unexpectedly nil")
 		}
 		ht.backfillMB.Deliver(headWithChain)
 		ht.broadcastMB.Deliver(headWithChain)
-	} else if head.Number == prevHead.Number {
-		if head.Hash != prevHead.Hash {
-			ht.log.Debugw("Got duplicate head", "blockNum", head.Number, "head", head.Hash.Hex(), "prevHead", prevHead.Hash.Hex())
+	} else if head.BlockNumber() == prevHead.BlockNumber() {
+		if head.BlockHash() != prevHead.BlockHash() {
+			ht.log.Debugw("Got duplicate head", "blockNum", head.BlockNumber(), "head", head.BlockHash(), "prevHead", prevHead.BlockHash())
 		} else {
-			ht.log.Debugw("Head already in the database", "head", head.Hash.Hex())
+			ht.log.Debugw("Head already in the database", "head", head.BlockHash())
 		}
 	} else {
-		ht.log.Debugw("Got out of order head", "blockNum", head.Number, "head", head.Hash.Hex(), "prevHead", prevHead.Number)
-		if head.Number < prevHead.Number-int64(ht.config.FinalityDepth()) {
+		ht.log.Debugw("Got out of order head", "blockNum", head.BlockNumber(), "head", head.BlockHash(), "prevHead", prevHead.BlockNumber())
+		prevUnFinalizedHead := prevHead.BlockNumber().Int64() - int64(ht.config.FinalityDepth())
```

Comment:
would appreciate comments on the naming for this! `prevUnFinalizedHead`  in `prevUnFinalizedHead := prevHead.BlockNumber().Int64() - int64(ht.config.FinalityDepth())`